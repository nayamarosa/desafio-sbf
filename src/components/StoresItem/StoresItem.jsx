import React from "react";

import Icon from "../Icon";

const StoresItem = ({ name, adress, onClick }) => {
  return (
    <li className="stores__list-item">
      <p className="stores__list-item--title">{name}</p>
      <a href="/" className="show-map" onClick={onClick}>
        <Icon icon='./images/icon_pin_link.svg' alt="Ver no mapa" />
        <p className="stores__list-item--map">Ver no mapa</p>
      </a>
      <div className="stores__list-item-group">
        <p className="stores__list-item--address">{adress}</p>
        <p className="stores__list-item--time">Atendimento:</p>
        <p className="stores__list-item--time">Segunda a Sábado 10h às 22h | Domingo 11h às 20h</p>
        <span>Disponível em 4 dias úteis</span>
      </div>
    </li>
  )
}

export default StoresItem;