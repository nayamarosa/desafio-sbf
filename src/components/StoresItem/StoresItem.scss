.stores__list {
  &-item {
    margin: 16px 0;
    padding: 0 16px;
    &--title {
      margin-bottom: 8px;
      @include font(24px, 500, auto);
    }
    &--map {
      margin-bottom: 16px;
      @include font(16px, 400, 1.5);
      text-decoration: underline;
    }
    &-group {
      padding: 24px;
      background-color: $white;
      color: $dark-gray;
      border-radius: 8px;
      & span {
        @include font(12px, 700, 1.5);
        color: #006611;
      }
    }
    &--address {
      margin-bottom: 8px;
      @include font(14px, 400, 1.5);
    }
    &--time {
      @include font(12px, 400, 1.5);
    }
  }
}