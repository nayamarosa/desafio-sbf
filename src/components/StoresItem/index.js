import StoresItem from "./StoresItem";

export default StoresItem;