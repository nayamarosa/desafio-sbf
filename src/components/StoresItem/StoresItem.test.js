import React from "react";
import { render } from "@testing-library/react";

import StoresItem from "./StoresItem";

describe('Store Item Component', () => {
  it('same snapshot component', () => {
    const { container } = render(<StoresItem 
      name="loja"
      adress="Rua tal, nÃºmero tal"
    />)
    
    expect(container.firstChild).toMatchSnapshot();
  })
})