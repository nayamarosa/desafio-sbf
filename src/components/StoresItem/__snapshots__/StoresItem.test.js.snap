// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Item Component same snapshot component 1`] = `
<li>
  <p>
    loja
  </p>
  <p>
    Rua tal, número tal
  </p>
  <p>
    horário de funcionamento
  </p>
</li>
`;
